homeassistant/components/switch/deluge.py

```diff
@@ -76,15 +79,34 @@ def is_on(self):
         """Return true if device is on."""
         return self._state == STATE_ON
 
+    @property
+    def available(self):
+        """Return true if device is available."""
+        return self._available
+
     def turn_on(self, **kwargs):
         """Turn the device on."""
-        self.deluge_client.call('core.resume_all_torrents')
+        # self.deluge_client.call('core.resume_all_torrents')
+        torrent_ids = self.deluge_client.call('core.get_session_state')
+        self.deluge_client.call('core.resume_torrent', torrent_ids)
 
     def turn_off(self, **kwargs):
         """Turn the device off."""
-        self.deluge_client.call('core.pause_all_torrents')
+        # self.deluge_client.call('core.pause_all_torrents')
+        torrent_ids = self.deluge_client.call('core.get_session_state')
+        self.deluge_client.call('core.pause_torrent', torrent_ids)
 
     def update(self):
+        """ check if deluge_client is connected first """
+        try:
+            self.deluge_client._create_socket()
```

Comment:
Originally the setup platform was trying to connect to the deluge daemon on startup and only create the switch if the connection is successful. this means that if the deluge is offline at the time we need to restart HA when deluge is back online to create the switch.

My idea is to create the switch regardless and set it as unavailable. when the update function is called we will make a new connection to the deluge daemon. if the connection is successful then we make the switch available and we can do the `.call(...)` function to update the status of the switch. this will also make the `self.deluge_client` available for the `turn_on()` and `turn_off()` functions.

But in case the deluge daemon is disconnected at some time after HA has already started the connection will be broken and the only way to reconnect is to create a new socket (which is initially created when running `deluge_api = DelugeRPCClient(host, port, username, password)` )

Reply:
Ok. What happens here on this line if the connection in `__init__` is successful? Will the client allow a new socket to be created and clean up the old socket?

I think the general approach you've described can work and is ok. You can also consider trying to connect inside `setup_platform` and raising `PlatfromNotReady` if connection fails. That will try to set up the platform again later. You also need to have reconnection logic in `update` as you say.